src/server/gc_worker/compaction_filter.rs

```diff
@@ -491,21 +537,33 @@ impl WriteCompactionFilter {
             let prefix = Key::from_encoded_slice(&self.mvcc_key_prefix);
             let def_key = prefix.append_ts(write.start_ts).into_encoded();
             self.write_batch.delete(&def_key)?;
-            self.flush_pending_writes_if_need()?;
         }
         Ok(())
     }
 
-    fn delete_write_key(&mut self, key: &[u8]) -> Result<(), String> {
-        self.write_batch.delete_cf(CF_WRITE, key)?;
-        self.flush_pending_writes_if_need()?;
-        self.deleted += 1;
-        Ok(())
-    }
+    fn flush_pending_writes_if_need(&mut self, force: bool) -> Result<(), String> {
+        if self.write_batch.is_empty() {
+            return Ok(());
+        }
 
-    fn flush_pending_writes_if_need(&mut self) -> Result<(), String> {
-        if self.write_batch.count() > DEFAULT_DELETE_BATCH_COUNT {
-            self.write_batch.write()?;
+        fn do_flush(db: &DB, wb: &RocksWriteBatch, wopts: &WriteOptions) -> Result<(), String> {
+            fail_point!("write_compaction_filter_flush_write_batch", true, |_| {
+                Err("Ingested fail point".to_owned())
+            });
+            db.write_opt(wb.as_inner(), wopts)
+        }
+
+        if self.write_batch.count() > DEFAULT_DELETE_BATCH_COUNT || force {
+            let db = self.engine.as_inner();
+            let mut wopts = WriteOptions::default();
+            wopts.set_no_slowdown(true);
```

Comment:
btw do we do sync write for GC?

Reply:
Yes, the write batch is sync written in GC worker.